File: app/src/main/java/com/nhn/fitness/ui/adapters/MessageDiffUtil.java
package com.nhn.fitness.ui.adapters;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.nhn.fitness.data.model.Message;

public class MessageDiffUtil extends DiffUtil.ItemCallback<Message> {

   @Override
   public boolean areItemsTheSame(@NonNull Message oldItem, @NonNull Message newItem) {
      return oldItem.getId() == newItem.getId();
   }

   @Override
   public boolean areContentsTheSame(@NonNull Message oldItem, @NonNull Message newItem) {
      return oldItem.getId() == newItem.getId();
   }
}

File: app/src/main/java/com/nhn/fitness/ui/adapters/chat/MessageDiffUtil.kt
package com.nhn.fitness.ui.adapters.chat

import androidx.recyclerview.widget.DiffUtil
import com.nhn.fitness.data.model.Message

class MessageDiffUtil: DiffUtil.ItemCallback<Message>() {
    override fun areItemsTheSame(oldItem: Message, newItem: Message): Boolean {
        return oldItem == newItem
    }

    override fun areContentsTheSame(oldItem: Message, newItem: Message): Boolean {
        return oldItem.id == newItem.id
    }
}